### PR TITLE
buildFHSEnv: disable multiArch by default

### PR DESCRIPTION
**Pull request by Atemu**
###### Description of changes

Most FHSEnv-wrapped packages in Nixpkgs wrap a x86_64-linux binary; making
multiArch unnecessary bloat closure size. Saves about 1G in anki-bin.

This makes multiArch FHSEnvs the exception rather than the rule.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSamus** [comment]
Heroic and Bottles also need it enabled (as Wine asks for 32 bit dependencies for 32 bit applications at the moment).

**Comment by Atemu** [comment]
cc All maintainers of at least one FHS-wrapped package:

@1000101 @aidalgol @antonxy @badmutex @bennofs @bhipple @bobby285271 @eadwu @emmanuelrosa @Enzime @euank @Forkk @GRBurst @hellwolf @huantianad @jluttine @joepie91 @LnL7 @lucasew @Madouura @maxeaubrey @MayNiklas @necessarily-equal @oxij @Patryk27 @PhilippWoelfel @pjones @Profpatsch @psydvl @rvolosatovs @SCOTT-HAMILTON @stepech @Synthetica9 @tesq0 @thoughtpolice @ttuegel @turion @urandom2 @vojta001 @Yureien @zimbatm

File: pkgs/games/heroic/fhsenv.nix [hunk 1]
```diff
@@ -9,6 +9,9 @@ buildFHSEnv {
 
   runScript = "heroic";
 
+  # Many WINE games need 32bit
```
Comment (aidalgol) [inline pkgs/games/heroic/fhsenv.nix:12]:
```suggestion
  # Many Wine and native games need 32-bit libraries.
```

**Review by aidalgol — APPROVED** [review]
Looks good.  I just suggested a minor correction/clarification in the comment in `heroic`'s `fhsenv.nix`.

**Comment by huantianad** [comment]
`unityhub` seems to work, launching, playing, and building a project on editor version 2022.3.3f1 seems to work fine. I'll test if older versions work with blank projects, but if anyone has projects on older versions, I would appreciate testing those as well.

**Comment by loco-choco** [comment]
I sure can test in some, what kind of difference are we supposed to see with this change?

**Comment by huantianad** [comment]
There shouldn't be any noticeable differences, just testing to make sure that the editor still works fine and that there's no issues with the editor, player, or building in editor.

**Comment by loco-choco** [comment]
Tested with two projects, built in one and compiled assetbundles in another, and didn't see any noticeable difference.

**Comment by bddvlpr** [comment]
Works fine for me on `2019.4.31f1`.

**Comment by emmanuelrosa** [comment]
I confirmed that the sparrow package doesn't need `multiPkgs`. The native code is 64-bit.

```
diff --git a/pkgs/applications/blockchains/sparrow/fhsenv.nix b/pkgs/applications/blockchains/sparrow/fhsenv.nix
index 26bf9da0b85..16ec980ab00 100644
--- a/pkgs/applications/blockchains/sparrow/fhsenv.nix
+++ b/pkgs/applications/blockchains/sparrow/fhsenv.nix
@@ -13,9 +13,7 @@ buildFHSEnv {
     pcsclite
   ];

-  multiPkgs = pkgs: with pkgs; [
-    pcsclite
-  ];
+  multiArch = false;

   extraInstallCommands = ''
     mkdir -p $out/share
```

**Comment by Atemu** [comment]
Last call, I'll merge this soon.

**Comment by SuperSamus** [comment]
Is there a reason to keep `multiArch = false` in many packages? (14)

Plus one that does this:
https://github.com/NixOS/nixpkgs/blob/0fa5a936f203acc1b11ed20fe002320944a8363b/pkgs/applications/misc/notesnook/default.nix#L50-L51

**Comment by Atemu** [comment]
@SuperSamus not that I know of other than to be explicit about it.

The `null` hack is no longer required/should be replaced by `multiArch`. I tried replacing all usages but may have missed one.

**Pull request merged**